lib/LaTeXML/MathParser.pm

```diff
@@ -1234,33 +1234,108 @@ sub extract_separators {
   '|' => '|', '||' => '||',
+  '<' => '>',
   "\x{230A}" => "\x{230B}",    # lfloor, rfloor
   "\x{2308}" => "\x{2309}",    # lceil, rceil
   "\x{2329}" => "\x{232A}",    # angle brackets; NOT mathematical, but balance in case they show up.
   "\x{27E8}" => "\x{27E9}",    # angle brackets (preferred)
   "\x{2225}" => "\x{2225}",    # lVert, rVert
+  "\x{2191}" => "\x{2193}",    # uArrow, dArrow
+  "\x{21D1}" => "\x{21D3}",    # uDoubleArrow, dDoubleArrow
+  "\x{2195}" => "\x{21D5}", # Test weirdness, is this real world relevant? "Up Down Arrow" and "Up Down Double Arrow"
 );
+my %fences = ();
+for my $key (keys(%balanced)) {
+  $fences{$key} = 1;
+  $fences{ $balanced{$key} } = 1;
+}
+
+our %notations = ();
+
+sub Notation {
+  my ($pattern, $meaning) = @_;
+  $notations{$pattern} = $meaning;    # KISS for now, should eventually use a match-builder ?
+  return;
+}
+
 # For enclosing a single object
 # Note that the default here is just to put open/closed attributes on the single object
-my %enclose1 = (    # [CONSTANT]
-  '{@}'   => 'set',                                   # alternatively, just variant parentheses
-  '|@|'   => 'absolute-value',
-  '||@||' => 'norm', "\x{2225}@\x{2225}" => 'norm',
-  "\x{230A}@\x{230B}" => 'floor',
-  "\x{2308}@\x{2309}" => 'ceiling',
-  '<@>'               => 'expectation',               # or just average?
-  '<@|'               => 'bra', '|@>' => 'ket');
+Notation('()',                'unwrap-delimited');
+Notation('(@)',               'unwrap-delimited');
+Notation('{@}',               'set');                # alternatively, just variant parentheses
+Notation('|@|',               'absolute-value');
+Notation('||@||',             'norm');
+Notation("\x{2225}@\x{2225}", 'norm');
+Notation("\x{230A}@\x{230B}", 'floor');
+Notation("\x{2308}@\x{2309}", 'ceiling');
+Notation('<@>',               'expectation');        # or just average;
+Notation('<@|',               'bra');
+Notation('|@>',               'ket');
 # For enclosing more than 2 objects; the punctuation is significant too
-my %enclose2 = (                                      # [CONSTANT]
-  '(@,@)' => 'open-interval',                                           # alternatively, just a list
-  '[@,@]' => 'closed-interval',
-  '(@,@]' => 'open-closed-interval', '[@,@)' => 'closed-open-interval',
-  '{@,@}' => 'set',                                                     # alternatively, just a list ?
-);
+Notation('(@,@)', 'open-interval');                  # alternatively, just a list
+Notation('[@,@]', 'closed-interval');
+Notation('(@,@]', 'open-closed-interval');
+Notation('[@,@)', 'closed-open-interval');
+Notation('{@,@}', 'set');                            # alternatively, just a list ?
+
 # For enclosing more than 2 objects.
 # assume 1st punct? or should we check all are same?
-my %encloseN = (    # [CONSTANT]
-  '(@,@)' => 'vector', '{@,@}' => 'set',);
+Notation('(@,@,@)', 'vector');
+Notation('{@,@,@}', 'set');
```

Comment:
These last 2 rules really wanted to be `\(@(,@)+\)` and `\{@(,@)+\}` instead, I am just punting here until I get the machinery in place.